modules/library/data/src/main/java/org/geotools/data/memory/MemoryFeatureStore.java

```diff
@@ -94,4 +95,17 @@ protected int getCountInternal(Query query) throws IOException {
     protected boolean handleVisitor(Query query, FeatureVisitor visitor) throws IOException {
         return delegate.handleVisitor(query, visitor);
     }
+
+    /**
+     *  GEOT-5683] preserve FeatureIDd during add feature
```

Comment:
I tend to only mention bug reports in test cases, GEOT-5683 will not add value to someone trying to learn how to use MemoryDataStore for the 1st time.